src/main/java/duke2/MainWindow.java

```diff
@@ -0,0 +1,86 @@
+package duke2;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.VBox;
+
+
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow {
+        @FXML
+        private ScrollPane scrollPane;
+        @FXML
+        private VBox dialogContainer;
+        @FXML
+        private TextField userInput;
+        @FXML
+        private Button sendButton = new Button("Send");
+
+        private Duke duke;
+
+        private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+        private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+        @FXML
+        public void initialize() {
+            scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        }
+
+        public void setDuke(Duke d) {
+            duke = d;
+        }
+
+        /**
+         * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+         * the dialog container. Clears the user input after processing.
+         */
+        @FXML
+        private void handleUserInput() {
+            String input = userInput.getText();
+            String response = duke.getResponse(input);
+            dialogContainer.getChildren().addAll(
+                    DialogBox.getUserDialog(input, userImage),
+                    DialogBox.getDukeDialog(response, dukeImage)
+            );
+            userInput.clear();
+        }
+
+//        scrollPane.setContent(dialogContainer);
```

Comment:
Hi Sophie, I am the one that reviewed your PR for the tutorial! Just sharing some of my thoughts that might help.

I feel that you should remove all these commented code because they are not used anymore and doesn't help when others are reading your code. Moreover, might confuse you as well! If you need to retrieve them in the future can always revisit the commits :)